src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverCountingVisitor.java

```diff
@@ -0,0 +1,54 @@
+package edu.kis.powp.jobs2d.drivers.visitor;
+
+import edu.kis.powp.jobs2d.drivers.*;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.jobs2d.drivers.transformators.TransformingJob2dDriverDecorator;
+
+public class DriverCountingVisitor implements IDriverVisitor {
+
+    private int count = 0;
+
+    public int countDrivers(IVisitableDriver driver) {
+        count = 0;
+        driver.accept(this);
+        return count;
+    }
+
+    @Override
+    public void visit(DriversComposite driversComposite) {
+        for(IVisitableDriver driver : driversComposite.getDrivers()) {
+            driver.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(LoggerDriver loggerDriver) {
+        count++;
+    }
+
+    @Override
+    public void visit(LineDriverAdapter lineDriverAdapter) {
+        count++;
+    }
+
+    @Override
+    public void visit(RealTimeDecoratorDriver realTimeDecoratorDriver) {
+        realTimeDecoratorDriver.getDriver().accept(this);
+    }
+
+    @Override
+    public void visit(UsageMonitorDriverDecorator usageMonitorDriverDecorator) {
+        usageMonitorDriverDecorator.getDriver().accept(this);
+    }
+
+    @Override
+    public void visit(RecordingDriverDecorator recordingDriverDecorator) {
+        recordingDriverDecorator.getDriver().accept(this);
+    }
+
+    @Override
+    public void visit(TransformingJob2dDriverDecorator transformingJob2dDriverDecorator) {
+        transformingJob2dDriverDecorator.getDriver().accept(this);
```

Comment:
decorators are also drivers. Here, I would also expect :
count++
Or visitor could be smarter and could count decorators and composites using separate counter:
decoratorsCount++
and
compositesCount++